pynicotine/privatechat.py

```diff
@@ -209,73 +189,113 @@ def _user_status(self, msg):
         if msg.status == slskmessages.UserStatus.OFFLINE:
             self.private_message_queue.pop(msg.user, None)
 
+    def get_message_type(self, text, is_outgoing_message):
+
+        if text.startswith("/me "):
+            return "action"
+
+        if is_outgoing_message:
+            return "local"
+
+        if core.login_username and find_whole_word(core.login_username.lower(), text.lower()) > -1:
+            return "hilite"
+
+        return "remote"
+
     def _message_user(self, msg, queued_message=False):
         """Server code 22."""
 
         username = msg.user
-        message = msg.msg
-
-        if not queued_message:
-            log.add_chat(_("Private message from user '%(user)s': %(message)s"), {
-                "user": username,
-                "message": message
-            })
-
-            core.send_message_to_server(slskmessages.MessageAcked(msg.msgid))
+        message = msg.message
+        print(message)
+        print("YYY")
+        timestamp = msg.timestamp if not msg.is_new_message else None
+        is_outgoing_message = (msg.message_id is None)
+
+        if not is_outgoing_message:
+            if not queued_message:
+                log.add_chat(_("Private message from user '%(user)s': %(message)s"), {
```

Comment:
Consider ditching this string and use `f"[{username}] {message}"` instead.

In future (and in the past I think), viewing the [Chat] log may well be how chat will (barely) operate in headless CLI.

Reply:
I'll leave this to post-3.3.0. Not sure the difference between chat room and private messages would be large enough.